music21/metadata/__init__.py

```diff
@@ -308,48 +743,281 @@ def all(self, skipContributors=False):
+        # search wants to find everything.
+
+        if attributeName in properties.UNIQUE_NAME_TO_NSKEY:
+            return self._getStringValuesByNSKey(properties.UNIQUE_NAME_TO_NSKEY[attributeName])
+
+        # Is attributeName a grandfathered workId?
+        if attributeName in properties.MUSIC21_WORK_ID_TO_NSKEY:
+            return self._getStringValuesByNSKey(properties.MUSIC21_WORK_ID_TO_NSKEY[attributeName])
+
+        # Is attributeName a grandfathered workId abbreviation?
+        if attributeName in properties.MUSIC21_ABBREVIATION_TO_NSKEY:
+            return self._getStringValuesByNSKey(
+                properties.MUSIC21_ABBREVIATION_TO_NSKEY[attributeName])
+
+        # The following are in searchAttributes, and getattr will find them because
+        # they are a property, but this routine needs to find them, too.
+        if attributeName == 'fileFormat':
+            if self.fileFormat is None:
+                return []
+            return [self.fileFormat]
+
+        if attributeName == 'filePath':
+            if self.filePath is None:
+                return []
+            return [self.filePath]
+
+        if attributeName == 'fileNumber':
+            if self.fileNumber is None:
+                return []
+            return [str(self.fileNumber)]
+
+        raise AttributeError(f'invalid attributeName: {attributeName}')
+
+    def _getSingularAttribute(self, attributeName: str) -> t.Optional[str]:
+        if attributeName in properties.UNIQUE_NAME_TO_NSKEY:
+            return self._getStringValueByNSKey(properties.UNIQUE_NAME_TO_NSKEY[attributeName])
+
+        # Is name a grandfathered workId?
+        if attributeName in properties.MUSIC21_WORK_ID_TO_NSKEY:
+            return self._getStringValueByNSKey(properties.MUSIC21_WORK_ID_TO_NSKEY[attributeName])
+
+        # Is name a grandfathered workId abbreviation?
+        if attributeName in properties.MUSIC21_ABBREVIATION_TO_NSKEY:
+            return self._getStringValueByNSKey(
+                properties.MUSIC21_ABBREVIATION_TO_NSKEY[attributeName])
+
+        raise AttributeError(f'object has no attribute: {attributeName}')
+
     def __getattr__(self, name):
-        r'''
-        Utility attribute access for attributes that do not yet have property
-        definitions.
-        '''
-        match = None
-        for abbreviation, workId in self.workIdAbbreviationDict.items():
-            # for id in WORK_IDS:
-            # abbreviation = workIdToAbbreviation(id)
-            if name == workId:
-                match = workId
-                break
-            elif name == abbreviation:
-                match = workId
-                break
-        if match is None:
-            raise AttributeError(f'object has no attribute: {name}')
-        result = self._workIds[match]
-        # always return string representation for now
-        return str(result)
+        '''
+        Utility attribute access for all uniqueNames, grandfathered workIds,
```

Comment:
I think that this is no longer just a `Utility attribute` access.

Reply:
Done.